platform/darwin/src/NSExpression+MGLAdditions.h

```diff
@@ -42,6 +90,133 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, readonly) id mgl_jsonExpressionObject;
 
+/**
+ Returns a constant expression containing an `NSString`.
+ 
+ This is equivalent to call `[NSExpression expressionForConstantValue:]`.
+ 
+ @param string The string constant.
+ */
++ (instancetype)mgl_expressionForString:(nonnull NSString *)string;
+
+/**
+ Returns a constant expression containing an `MGLColor`.
+ 
+ This is equivalent to call `[NSExpression expressionForConstantValue:]`.
+ 
+ @param color The color constant.
+ */
++ (instancetype)mgl_expressionForColor:(nonnull MGLColor *)color;
+
+/**
+ Returns a constant expression containing an `NSValue`.
+ 
+ This is equivalent to call `[NSExpression expressionForConstantValue:]`.
+ 
+ @param value The value constant.
+ */
++ (instancetype)mgl_expressionForValue:(nonnull NSValue *)value;
+
+/**
+ Returns a conditional function expression specifying the string predicate, and
+ colors for each condition.
+ 
+ @param conditionPredicate The predicate to get evaluated.
+ @param trueColor The color for conditions equal to true.
+ @param falseColor The color value if the condition is equal to false.
+ */
++ (instancetype)mgl_expressionForConditional:(nonnull NSString *)conditionPredicate trueColor:(nonnull MGLColor*)trueColor falseColor:(nonnull MGLColor *)falseColor;
+
+/**
+ Returns a conditional function expression specifying the string predicate, and
+ values for each condition.
+ 
+ @param conditionPredicate The predicate to get evaluated.
+ @param trueValue The value for conditions equal to true.
+ @param falseValue The value for conditions equal to false.
+ */
++ (instancetype)mgl_expressionForConditional:(nonnull NSString *)conditionPredicate trueValue:(nonnull NSValue *)trueValue falseValue:(nonnull NSValue *)falseValue;
+
+/**
+ Returns a conditional function expression specifying the string predicate, and
+ expressions for each condition.
+ 
+ @param conditionPredicate The predicate to get evaluated.
+ @param trueExpression The expression for conditions equal to true.
+ @param falseExpression The expression for conditions equal to false.
+ */
++ (instancetype)mgl_expressionForConditional:(nonnull NSString *)conditionPredicate trueExpression:(nonnull NSExpression *)trueExpression falseExpresssion:(nonnull NSExpression *)falseExpression;
```

Comment:
Can we rename this method `+expressionForConditional:trueExpression:falseExpression:` and conditionalize it for iOS and macOS versions that lack this initializer?

/ref https://github.com/mapbox/mapbox-gl-native/pull/11278#discussion_r172640706

Reply:
Actually, that won’t work, because I don’t think the method is implemented at all on older versions of iOS and macOS – it isn’t just missing from the public headers.

Maybe we can hold off on this initializer until #11007 is implemented. For that issue, we need an initializer that can take an arbitrary number of conditional predicates and expressions, which will probably entail a vararg syntax.